### ISSUE TITLE
Duplicated edit controls

### ISSUE THREAD

**Issue by janekolszak** [alex3165/react-leaflet-draw#150]
Hi!
I'm getting two sets of edit controls, like in the screenshot:
![image](https://user-images.githubusercontent.com/3087849/175825315-4be37885-fed8-4192-90e2-05cc01dfe3e6.png)

```typescript
import { Ref, useEffect, useMemo, useRef, useState } from 'react';

import Leaflet, { LatLngLiteral, rectangle } from 'leaflet';
import { LayersControl, Circle, FeatureGroup, TileLayer, Marker, Popup, Tooltip, Polygon, MapContainer } from 'react-leaflet';

import 'leaflet/dist/leaflet.css';
import 'leaflet-draw/dist/leaflet.draw.css'
import { EditControl } from 'react-leaflet-draw';


interface Props {
  value: GeoJSON.FeatureCollection<any>;
  setValue: (item: GeoJSON.FeatureCollection<any>) => void;
}


const center = {
  lat: 52.2297,
  lng: 21.0122,
}

const style = {
  color: '#ee7219',
  weight: 1,
  opacity: 1,
  fillOpacity: 0.5,
  fillColor: '#ee7219'
};

const LocationPolygonPicker: React.FunctionComponent<Props> = (props) => {
  const editRef = useRef<EditControl>(null)
  let featureGroup: Leaflet.FeatureGroup

  const onCreated = e => {
    const layers = featureGroup?.getLayers()
    if (layers && layers.length > 1) {
      // console.log("There was a layer", featureGroup?.toGeoJSON())
      layers.forEach((layer, idx) => {
        if (idx !== layers.length - 1) {
          layer.remove()
        }
      })
    }
  };

  const onFeatureGroupReady = (fg: Leaflet.FeatureGroup) => {
    if (!fg) {
      return
    }
    fg.clearLayers()
    Leaflet.geoJSON(props.value, { style: style }).eachLayer(layer => {
      fg.addLayer(layer)
    })
    featureGroup = fg
  }

  return (
    <>
      <MapContainer
        center={center}
        zoom={13}
        scrollWheelZoom={false}
        style={{ height: 390, width: '100%' }}
        className="rounded-sm border-border-color"
      >
        <TileLayer
          tileSize={512}
          minZoom={1}
          zoomOffset={-1}
          attribution={process.env.NEXT_PUBLIC_MAP_ATTRIBUTION}
          url={process.env.NEXT_PUBLIC_MAP_URL!}
        />

        <FeatureGroup ref={onFeatureGroupReady}  >
          <EditControl
            ref={editRef}
            position='topright'
            onCreated={onCreated}
            onEdited={() => { console.log("onEdited") }}
            onDeleted={() => { console.log("onDeleted") }}
            onMounted={() => {
              console.log("onMounted")
              editRef.current?.forceUpdate()
            }}
            onEditStart={() => { console.log("onEditStart") }}
            onEditStop={() => { console.log("onEditStop") }}
            onDeleteStart={() => { console.log("onDeleteStart") }}
            onDeleteStop={() => { console.log("onDeleteStop") }}
            draw={{
              // polygon: { showArea: true, showLength: true, precision: { km: 2 } },
              polygon: {
                allowIntersection: false,
                shapeOptions: style,
                edit: false,
                showLength: true,
                metric: true,
                feet: false,
                showArea: true
              },
              circle: {
                allowIntersection: false,
                shapeOptions: style,
                edit: false,
                showLength: true,
                metric: true,
                feet: false,
                showArea: true
              },
              rectangle: false,
              marker: false,
              polyline: false,
              circlemarker: false
            }}
          />
        </FeatureGroup>
      </MapContainer>
    </>
  );
};

export default LocationPolygonPicker;
```

**Comment by brennanwilkes** [comment]
Hi @janekolszak, I'm looking into this issue and will fix it in v0.20.2, but in the meantime this css hack should help you: 
`
.leaflet-control ~ .leaflet-control{
  display: none;
}
`

**Comment by janekolszak** [comment]
Thanks!

**Comment by a-gradina** [comment]
> Hi @janekolszak, I'm looking into this issue and will fix it in v0.20.2, but in the meantime this css hack should help you:
> 
> ```
> .leaflet-control ~ .leaflet-control{
>   display: none;
> }
> ```

The CSS hack didn't work for us unfortunately. We are using v0.20.1.

If it helps anyone, this did the trick:

```css
.leaflet-control:first-of-type + .leaflet-control{
  display: none;
}
```

**Comment by Kreachers** [comment]
It would appear that it's linked to React.StrictMode causing useState, useEffect, and UseReducer functions to be [called twice](https://reactjs.org/docs/strict-mode.html#detecting-unexpected-side-effects). Removing React.StrictMode in my project fixed the duplicate controls. Not sure what a fix for this would be, Maybe debouncing the hook functions that create the controls.
It also looks like React.StrictMode doesn't take effect in a [production environment](https://github.com/facebook/react/issues/16295#issuecomment-610098654). So I guess if you aren't worries about duplicate controls on your development environment than its not really a problem.

**Comment by manuelpoelzl** [comment]
I have the same behavior, and get two edit controls as well, it's good to see that a fix is being worked on with v0.20.2!

This is definitely related to the new behavior of the strict mode in React18, since in a production build, only one control is rendered.

**Comment by brennanwilkes** [comment]
@Kreachers @manuelpoelzl In my testing, PR #159 solves this issue. Before I release, anything stand out to you as problematic?

**Comment by manuelpoelzl** [comment]
I just took a look at it, I don't see anything that should cause any problems now.

**Comment by Kreachers** [comment]
@brennanwilkes Yeah, from my limited react knowledge, it looks good to me.

**Comment by brennanwilkes** [comment]
Should be resolved with version [v0.20.2](https://www.npmjs.com/package/react-leaflet-draw/v/0.20.2)
